msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -1508,14 +1508,18 @@ private AccountRecord selectAccountRecordForTokenRequest(
             final boolean isUuid = isUuid(tenantId);
 
             if (!isUuid && !isHomeTenantAlias(tenantId)) {
-                tenantId = ((AzureActiveDirectoryAuthority) authority).getAudience().getTenantUuidForAlias();
+                tenantId = ((AzureActiveDirectoryAuthority) authority)
+                        .getAudience()
+                        .getTenantUuidForAlias(
+                                authority.getAuthorityURL().toString()
+                        );
             }
 
             IAccount accountForRequest;
 
             if (isHomeTenantAlias(tenantId)
                     || isAccountHomeTenant(multiTenantAccount.getClaims(), tenantId)) {
-                accountForRequest = multiTenantAccount;
+                accountForRequest = (multiTenantAccount.getClaims() != null) ? multiTenantAccount : null;
```

Comment:
if claims is null, then isAccountHomeTenant will return false.

Reply:
yes but it's an `or` check , so if  `isHomeTenantAlias` is true we end up setting an account with no claims